call-home/src/main.rs

```diff
@@ -0,0 +1,124 @@
+use std::{thread, time};
+pub mod collector;
+pub mod common;
+use clap::{App, Arg};
+use tokio;
+use sha256::digest;
+use crate::collector::http_client::HttpClient;
+use crate::collector::k8s_client::K8sClient;
+use crate::collector::report_models::{Pools, Replicas, Report, Volumes};
+use tracing::{info, Level, error,warn};
+use tracing_subscriber::FmtSubscriber;
+
+const PRODUCT: &str = common::constants::PRODUCT;
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    tracing_subscriber::fmt()
+        .with_max_level(tracing::Level::TRACE)
+        .init();
+
+    let matches = App::new(clap::crate_description!())
+        .version(clap::crate_version!())
+        .settings(&[
+            clap::AppSettings::ColoredHelp,
+            clap::AppSettings::ColorAlways,
+        ])
+        .arg(
+            Arg::with_name("endpoint")
+                .long("endpoint")
+                .short('e')
+                .default_value("http://mayastor-api-rest:8081")
+                .help("an URL endpoint to the control plane's rest endpoint"),
+        )
+        .arg(
+            Arg::with_name("namespace")
+                .long("namespace")
+                .short('n')
+                .default_value("mayastor")
+                .help("the default namespace we are supposed to operate in"),
+        )
+        .get_matches();
+    let namespace = matches.value_of("namespace").map(|s| s.to_string()).unwrap();
+    let endpoint= matches.value_of("endpoint").unwrap();
+    let version = clap::crate_version!().to_string();
+
+    let k8s_client = K8sClient::new().await.unwrap();
+    let http_client = HttpClient::new(endpoint).unwrap();
+
+    let mut report =generate_report(k8s_client.clone(), http_client.clone()).await;
+    report.deploy_namespace = namespace.clone();
+    report.product_version = version.clone();
+
+
+    loop{
+        // TODO: For now it loops every 60 sec. Need to change this to 24hr and set the value in constants
+        let time_to_sleep = time::Duration::from_secs(60);
+        thread::sleep(time_to_sleep);
```

Comment:
This will block the executor, it should be an async sleep, like [this one](https://docs.rs/tokio/latest/tokio/time/fn.sleep.html)